core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala

```diff
@@ -32,18 +33,56 @@ private[spark] case class Heartbeat(
     taskMetrics: Array[(Long, TaskMetrics)], // taskId -> TaskMetrics
     blockManagerId: BlockManagerId)
 
+private[spark] case object ExpireDeadHosts 
+    
 private[spark] case class HeartbeatResponse(reregisterBlockManager: Boolean)
 
 /**
  * Lives in the driver to receive heartbeats from executors..
```

Comment:
We should add a header comment that communicates that one of the functions of the HeartbeatReceiver is to expire executors.